samples/README.md

```diff
@@ -41,6 +41,69 @@ for more detailed instructions.
     pip install -r requirements.txt
     ```
 
+
+## Running tests locally
+
+Before running the tests, make sure you've followed the steps outlined in
+[Setup](#setup).
+
+### Install nox
+
+We use [nox](https://nox.readthedocs.io/en/latest/) to instrument our tests.
+
+```
+pip install nox
+```
+
+### Set environment variables
+
+You can run tests locally using your own gcs project or with a valid service account in project `python-docs-samples-tests`. This outlines the workflow of running tests locally using your own gcs project.
+
+Refer to [`noxfile_config.py`](https://github.com/googleapis/python-storage/blob/main/samples/snippets/noxfile_config.py) and [a list of environment variables](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/testing/test-env.tmpl.sh) that can be set manually. Not every test needs all of these variables.
+Below outlines some common environment variables used in the storage samples.
+See [Other Resources](#other-resources) on how to create credentials, keys, and secrets.
+
+    export GOOGLE_CLOUD_PROJECT=[your-project-name]
+    export MAIN_GOOGLE_CLOUD_PROJECT=[your-project-name]
+    export BUILD_SPECIFIC_GCLOUD_PROJECT=[your-project-name]
+    export HMAC_KEY_TEST_SERVICE_ACCOUNT=[your-service-account]
+    export CLOUD_KMS_KEY=[your-kms-key]
+    export GOOGLE_APPLICATION_CREDENTIALS=[your-credentials]
+
+If you are running a single test locally that does not use the environment variables, you can delete the `noxfile_config.py` file and simply set your `GOOGLE_CLOUD_PROJECT`
+
+```
+export GOOGLE_CLOUD_PROJECT=[your-project-name]
+```
+
+
+### Run tests with nox
+```
+nox -s lint
+nox -s py-3.7 -- snippets_test.py
```

Comment:
Now that 3.7 is well out of date, let's update this to 3.9

Reply:
good catch, updated